sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -457,16 +462,48 @@ export class StudentService extends RecordDataModelService<Student> {
         "user.firstName",
         "user.lastName",
         "sinValidation.sin",
-      ])
-      .leftJoin(
+      ]);
+    if (institutionId) {
+      searchQuery
+        .innerJoin(
+          Application,
+          "application",
+          "application.student.id = student.id",
+        )
+        .leftJoin("application.location", "pirLocation")
```

Comment:
Yes, I'm changing the submit logic to save the location in a different PR.

Reply:
> Are we changing the submit logic to save the location in a different PR?
> 
> And if that is going to happen, this whole part can be removed right?
> 
> ```
>         .leftJoin("application.currentAssessment", "studentAssessment")
>         .leftJoin("studentAssessment.offering", "offering")
>         .leftJoin("offering.institutionLocation", "offeringLocation")
>         .leftJoin("offeringLocation.institution", "offeringInstitution");
> ```

During previous conversations, we had in mind doing both checks, but since we will save the `application.location` during the application submission and assuming that the location can only be changed by editing the application, it could be a nice simplification of the query.